spring-xd-dirt/src/main/java/org/springframework/xd/dirt/cluster/ContainerFilter.java

```diff
@@ -28,6 +28,6 @@
  */
 public interface ContainerFilter {
 
-	List<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
+	Collection<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
```

Comment:
The contract for this method implies that
* containers are provided in a specific order (via `Iterable`)
* the order in which containers are returned is not in any particular order (since it returns `Collection`)

If the filter is responsible for maintaining order, then it should return `List`.

Alternatively, what do you think about making the filter return a boolean, given the input of `ModuleDescriptor` and `Container`? This would allow for a single iteration of containers within `ContainerMatcher` instead of having to iterate all of the containers for each provided filter.